e2e_playwright/st_form_container_association_test.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from playwright.sync_api import Page
+
+from e2e_playwright.conftest import wait_for_app_run
+from e2e_playwright.shared.app_utils import (
+    click_checkbox,
+    click_form_button,
+    expect_prefixed_markdown,
+)
+
+
+def test_form_container_association(app: Page):
+    """Test association of elements to a form with different cases
+    with elements created in the form container and outside.
+    """
+
+    # Form id mapping whether a checkbox is inside a form
+    checkbox_inside_form = {
+        "form_0": True,
+        "form_1": False,
+        "form_2": False,
+        "form_3": True,
+        "form_4": True,
+        "form_5": False,
+        "form_6": True,
+        "form_7": True,
+        "form_8": True,
+        "form_9": True,
+    }
+
+    for form_id, is_checkbox_inside_form in checkbox_inside_form.items():
```

Comment:
I guess it's better to use `pytest.parameterize` instead of a loop inside the test function, because then its easier to figure out what test failed when one of them fails (aka the reporting is better).

Reply:
I fully agree that this would be the better setup for normal unit tests. But in the case of the playwright, there is some overhead to consider when it's split up into 30 test runs instead of 3 (10* 3 browsers). If I have some time later, I can test how much difference it makes.

Reply:
I tested around a bit. Locally it takes 2x longer and on CI as well. But CI currently can fully parallize it. So, if there are other tests, the impact might be even bigger in the CI. I think it wouldn't too problematic to use this, but I also don't feel super well with it..

Reply:
I tested around a bit. Locally, it takes twice as long, and on CI as well. But CI in the changed tests run can fully parallelize it. So, if there are would be other tests, the impact might be even bigger in the CI. I think it wouldn't be too problematic to use this, but I also don't feel super well with it...

Reply:
I reverted it for now and left a comment to explain the reasoning. The playwright CI job is already creeping up to the 15min mark...